### ISSUE TITLE
CodeDeployAgent auto update stops an in progress deployment

### ISSUE THREAD

**Issue by chriskinsman** [aws/aws-codedeploy-agent#78]
The code deploy agent decides to update itself.

It checks if a deploy is running
It downloads the new package
It stops the agent

Between the time it checks if a deploy is running and stops the agent a deploy has started.  This deploy then hangs at a random step due to the agent quitting.  It never starts back up again due to a different ticket #77  we logged regarding the update failing.

**Comment by chriskinsman** [comment]
I have heard from support this is a side affect of us running on ubuntu and dpkg unconditionally re-installing the agent regardless of package version.

I would encourage you to add a version check to the cron script to only update if the version has changed AND to reduce the frequency of auto-update from once an hour to once a day.

Thanks!

**Comment by niclic** [comment]
We ran into this issue when we started using CodeDeploy on Ubuntu, and were advised to simply remove the cron job that attempts to update the agent.

We install the agent in ec2 user data, and run the following command at the end:

`rm /etc/cron.d/codedeploy-agent-update`

This was in early 2015. I recently contacted support and they advised keeping it like this. It's not ideal but I'd rather have working deployments. Newly launched instances get the latest agent.
